client/lib/user-settings/index.js

```diff
@@ -204,10 +221,11 @@ UserSettings.prototype.getSetting = function( settingName ) {
 	var setting = null;
 
 	// If we haven't fetched settings, or if the setting doesn't exist return null
-	if ( this.settings && 'undefined' !== typeof this.settings[ settingName ] ) {
-		setting = ( 'undefined' !== typeof this.unsavedSettings[ settingName ] )
-			? this.unsavedSettings[ settingName ]
-			: this.settings[ settingName ];
+	if ( this.settings && 'undefined' !== typeof get( this.settings, settingName ) ) {
```

Comment:
Yes, that should work. 👍  
Just without the `!`, as it's checking for not undefined:
 `if ( has( this.settings, settingName ) ) {`